components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/resources/publisher-api.yaml

```diff
@@ -2237,6 +2237,201 @@ paths:
             Requested API does not exist.
           schema:
             $ref: '#/definitions/Error'
+
+######################################################
+# The "Endpoint Collection" resource API
+######################################################
+  /endpoints:
+
+#-----------------------------------------------------
+# Retrieve the list of endpoints
+#-----------------------------------------------------
+    get:
+      x-scope: apim:api_view
+      x-wso2-curl: "curl -k -H \"Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8\" \"https://127.0.0.1:9443/api/am/publisher/v0.10/endpoints\""
+      x-wso2-request: |
+        GET https://127.0.0.1:9443/api/am/publisher/v0.10/endpoints
+        Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8
+      x-wso2-response: "HTTP/1.1 200 OK\nContent-Type: application/json\n\n{\n   \"list\": [   {\n      \"showInApiConsole\": true,\n      \"serverUrl\": \"https://192.168.56.1:9444//services/\",\n      \"endpoints\":       {\n         \"http\": \"http://192.168.56.1:8281\",\n         \"https\": \"https://192.168.56.1:8244\"\n      },\n      \"name\": \"Production and Sandbox\",\n      \"type\": \"hybrid\"\n   }],\n   \"count\": 1\n}"
+      summary: Get all endpoints
+      description: |
+        This operation can be used to retrieve the list of endpoints available.
+      tags:
+        - Endpoint (Collection)
+      responses:
+        200:
+          description: |
+            OK.
+            Endpoint list is returned.
+          schema:
+            type: array
```

Comment:
Shall we define this as a List model like EnvironmentList. Then we can specify other attributes like "count"  and pagination related attributes (if required)